### PR TITLE
Bug/jonas rut/refactor main

### PR DESCRIPTION
**Pull request by JonasRut**
ChargingController?

### CODE REVIEW

File: include/charge_controller/ChargeController.hpp [hunk 1]
```diff
@@ -135,6 +141,12 @@ class ChargeController {
     BMSManager& bms;
     LCDDisplay& display;
     IO::CAN& can;
+    DEV::Button& startButton;
+    IO::GPIO* statusLED;
+    IO::UART* uart;
```
Comment (mjmagee991) [inline include/charge_controller/ChargeController.hpp]: This class shouldn't have a pointer to UART in it

File: include/charge_controller/ChargeController.hpp [hunk 2]
```diff
@@ -135,6 +141,12 @@ class ChargeController {
     BMSManager& bms;
     LCDDisplay& display;
     IO::CAN& can;
+    DEV::Button& startButton;
+    IO::GPIO* statusLED;
```
Comment (mjmagee991) [inline include/charge_controller/ChargeController.hpp]: This should be a reference, not a pointer

File: src/charge_controller/ChargeController.cpp [hunk 3]
```diff
@@ -203,7 +246,7 @@ void ChargeController::sendChargerMessage() {
 
     if (state == ControllerStates::CHARGING) {
         // Multiply by ten to get the right sized values
-        voltage = 48 * 10;
+        voltage = 96 * 10;// This is for charging in a series
```
Comment (mjmagee991) [inline src/charge_controller/ChargeController.cpp]:
```suggestion
        voltage = 96 * 10;// This is for charging in series
```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
A little more cleanup

**Comment by mjmagee991** [comment]
[Charge Controller Main Loop Cleanup](https://trello.com/c/HuXDemVr/159-charge-controller-main-loop-cleanup)

File: include/charge_controller/ChargeController.hpp [hunk 4]
```diff
@@ -4,10 +4,14 @@
 #include <EVT/io/CAN.hpp>
 #include <EVT/io/CANopen.hpp>
 
+#include "EVT/dev/button.hpp"
```
Comment (mjmagee991) [inline include/charge_controller/ChargeController.hpp]:
```suggestion
#include <EVT/dev/button.hpp>
```

File: src/charge_controller/ChargeController.cpp [hunk 5]
```diff
@@ -48,43 +48,85 @@ void ChargeController::process() {
...
+            startCharging();
+        }
+    }
+
+    if (bms.numConnected() != oldCount) {
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "%d batteries connected", bms.numConnected());
+        oldCount = bms.numConnected();
+    }
+
+    log::LOGGER.log(log::Logger::LogLevel::DEBUG, "works part1 ?");
```
Comment (mjmagee991) [inline src/charge_controller/ChargeController.cpp]:
```suggestion
```

File: src/charge_controller/ChargeController.cpp [hunk 6]
```diff
@@ -48,43 +48,85 @@ void ChargeController::process() {
...
+            break;
+        case IO::GPIO::State::HIGH:
+            statusLED.writePin(IO::GPIO::State::LOW);
+            break;
         }
+
+        // Need to send heartbeat can message to the charger
+        sendChargerMessage();
+        lastHeartBeat = time::millis();
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "works part2 ?");
```
Comment (mjmagee991) [inline src/charge_controller/ChargeController.cpp]:
```suggestion
```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
A couple details

**Review by mjmagee991 — APPROVED** [review]
To avoid having to re-review, I'm just going to approve, but please clean up the stuff above

**Comment by aclowmclaughlin** [comment]
I don't know if this is an issue but the entire checkBMS method is commented out because it was 
"faulting on bms.getMaxCellVoltage(i)
     max voltage is 12000
     bms.getMaxCellVoltage(i) returned 25871"
(from comment).

Is this something that needs to be fixed, or is it ok?

File: src/charge_controller/ChargeController.cpp [hunk 7]
```diff
@@ -48,44 +48,83 @@ void ChargeController::process() {
...
+        sendChargerMessage();
+        lastHeartBeat = time::millis();
+    }
+}
+
+uint8_t ChargeController::checkBMS() {
+    uint8_t status = 0;
+
+    /*
+    TODO: was faulting on bms.getMaxCellVoltage(i)
```
Comment (ActuallyTaylor) [inline src/charge_controller/ChargeController.cpp:87]: Why is this all commented out?

**Review by ActuallyTaylor — APPROVED** [review]
Looks good, but I have a comment about the checkBMS function

File: src/charge_controller/ChargeController.cpp (hunk 7 above)
Comment (mjmagee991) [reply src/charge_controller/ChargeController.cpp:87]: I talked to Jonas about this. Basically, all of the CAN stuff is broken, so I'd rather have it handled in a separate PR

**Comment by mjmagee991** [comment]
> I don't know if this is an issue but the entire checkBMS method is commented out because it was "faulting on bms.getMaxCellVoltage(i) max voltage is 12000 bms.getMaxCellVoltage(i) returned 25871" (from comment).
> 
> Is this something that needs to be fixed, or is it ok?

It does need to be fixed but not in this PR

**Pull request merged**